plugins/inputs/amd_rocm_smi/amd_rocm_smi_parser.conf

```diff
@@ -0,0 +1,27 @@
+data_format = "xpath_json"
+metric_name = "amd_rocm_smi"
+
+[[xpath_config]]
+  metric_selection = "//*[starts-with(name(.), 'card')]"
+  ignore_nan = true
+
+  [xpath_config.tags]
+    name = "name()"
+    gpu_id = "child::*[name() = 'GPU ID']"
+    gpu_unique_id = "child::*[name() = 'Unique ID']"
+
+  [xpath_config.fields]
+    temperature_sensor_edge = "number(child::*[name() = 'Temperature (Sensor edge) (C)'])"
+    temperature_sensor_junction = "number(child::*[name() = 'Temperature (Sensor junction) (C)'])"
+    temperature_sensor_memory = "number(child::*[name() = 'Temperature (Sensor memory) (C)'])"
+    power_draw = "number(child::*[name() = 'Average Graphics Package Power (W)'])"
+    driver_version = "/system/child::*[name() = 'Driver version']"
+
+  [xpath_config.fields_int]
+    fan_speed = "child::*[name() = 'Fan speed (%)']"
+    memory_total = "child::*[name() = 'VRAM Total Memory (B)']"
+    memory_used = "child::*[name() = 'VRAM Total Used Memory (B)']"
+    utilization_gpu = "child::*[name() = 'GPU use (%)']"
+    utilization_memory = "child::*[name() = 'GPU memory use (%)']"
+    clocks_current_sm = "substring(child::*[name() = 'sclk clock speed:'], 2, string-length(.)-5)"
+    clocks_current_memory = "substring(child::*[name() = 'mclk clock speed:'], 2, string-length(.)-5)"
```

Comment:
As someone relatively new to Telegraf, how would I know how to write these conf files? 

While having explicit variables in the plugin code looks lengthy, it is a clear pattern that contributors can look to and expect. This conf file is not something I think I could throw together quickly.

At the very least, is it worth having a doc file on what this file is and how you translated from JSON to this?

Reply:
Hey @powersj thanks for looking into this! The config file is a usual [xpath](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/xpath) config with a few minor adaptations, i.e. using `xpath_config` instead of the usual plugin prefix (e.g. `inputs.file.xpath`).

Where would you like me to document this? In each plugin that is transferred to the new general implementation or in the general implementation? The problem is, that the general implementation allows _any_ parser type...

Is it ok for you if I add some comments to that config file saying "hey look at the parser docu and use the prefix `xpath_config`"?